addon/components/horizontal-bar-chart.js

```diff
@@ -306,7 +311,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
       .text((d) => this.get('formatLabelFunction')(d.value))
       .attr(this.get('valueLabelAttrs'));
 
-    var labelWidth = this.get('labelWidth');
+    var labelWidth;
+    // If the chart contains a mix of negative and positive values, the axis
+    // and labels are in the middle of the chart, not at the edges of the chart,
+    // so allow more space to compute how the label is trimmed.
+    if (this.get('hasNegativeValues') && this.get('hasPositiveValues')) {
+      labelWidth = this.get('outerWidth') / 2;
+    } else {
+      labelWidth = this.get('labelWidth');
+    }
```

Comment:
@philpee2 after this change, the "Label Width" control will have no effect on the charts with both negative and positive values. That could be confusing to the end user.